Event Hub name does not allow underscores

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version
0.11.7
<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* azurerm_eventhub

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
terraform {
}

provider "azurerm" {
}

resource "azurerm_eventhub" "test_events_v0" {
  name                = "my_hub"
  resource_group_name = "my-group"
  namespace_name      = "mynamespace"
  partition_count     = "2"
  message_retention   = "3"
}
```

### Debug Output

```
2018/08/15 11:58:07 [INFO] Terraform version: 0.11.7
2018/08/15 11:58:07 [INFO] Go runtime version: go1.10.1
2018/08/15 11:58:07 [INFO] CLI args: []string{"/usr/local/Cellar/terraform/0.11.7/bin/terraform", "validate"}
2018/08/15 11:58:07 [DEBUG] Attempting to open CLI config file: /Users/ducas/.terraformrc
2018/08/15 11:58:07 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2018/08/15 11:58:07 [INFO] CLI command args: []string{"validate"}
2018/08/15 11:58:07 [DEBUG] checking for provider in "."
2018/08/15 11:58:07 [DEBUG] checking for provider in "/usr/local/Cellar/terraform/0.11.7/bin"
2018/08/15 11:58:07 [DEBUG] checking for provider in ".terraform/plugins/darwin_amd64"
2018/08/15 11:58:07 [DEBUG] found provider "terraform-provider-azurerm_v1.12.0_x4"
2018/08/15 11:58:07 [DEBUG] found valid plugin: "azurerm", "1.12.0", "/Users/ducas/Code/EventHub/test/.terraform/plugins/darwin_amd64/terraform-provider-azurerm_v1.12.0_x4"
2018/08/15 11:58:07 [DEBUG] checking for provisioner in "."
2018/08/15 11:58:07 [DEBUG] checking for provisioner in "/usr/local/Cellar/terraform/0.11.7/bin"
2018/08/15 11:58:07 [DEBUG] checking for provisioner in ".terraform/plugins/darwin_amd64"
2018/08/15 11:58:07 [DEBUG] New state was assigned lineage "89b0b165-7332-7572-e7ab-c010992f9d57"
2018/08/15 11:58:07 [INFO] terraform: building graph: GraphTypeValidate
2018/08/15 11:58:07 [DEBUG] Resource state not found for "azurerm_eventhub.test_events_v0": azurerm_eventhub.test_events_v0
2018/08/15 11:58:07 [TRACE] Graph after step *terraform.AttachStateTransformer:

azurerm_eventhub.test_events_v0 - *terraform.NodeValidatableResource
2018/08/15 11:58:07 [TRACE] Graph after step *terraform.RootVariableTransformer:

azurerm_eventhub.test_events_v0 - *terraform.NodeValidatableResource
2018/08/15 11:58:07 [TRACE] ProviderConfigTransformer: Starting for path: []
2018/08/15 11:58:07 [DEBUG] resource azurerm_eventhub.test_events_v0 using provider provider.azurerm
2018/08/15 11:58:07 [DEBUG] ReferenceTransformer: "azurerm_eventhub.test_events_v0" references: []
2018/08/15 11:58:07 [DEBUG] ReferenceTransformer: "provider.azurerm" references: []
2018/08/15 11:58:07 [DEBUG] Starting graph walk: walkValidate
2018-08-15T11:58:07.702+1000 [DEBUG] plugin: starting plugin: path=/Users/ducas/Code/EventHub/test/.terraform/plugins/darwin_amd64/terraform-provider-azurerm_v1.12.0_x4 args=[/Users/ducas/Code/EventHub/test/.terraform/plugins/darwin_amd64/terraform-provider-azurerm_v1.12.0_x4]
2018-08-15T11:58:07.712+1000 [DEBUG] plugin: waiting for RPC address: path=/Users/ducas/Code/EventHub/test/.terraform/plugins/darwin_amd64/terraform-provider-azurerm_v1.12.0_x4
2018-08-15T11:58:07.722+1000 [DEBUG] plugin.terraform-provider-azurerm_v1.12.0_x4: plugin address: timestamp=2018-08-15T11:58:07.721+1000 address=/var/folders/05/jzcw1_q924d3mxf7fg_gl_3m0000gn/T/plugin815612218 network=unix
2018/08/15 11:58:07 [DEBUG] Resource state not found for "azurerm_eventhub.test_events_v0": azurerm_eventhub.test_events_v0
2018/08/15 11:58:07 [DEBUG] ReferenceTransformer: "azurerm_eventhub.test_events_v0" references: []
2018/08/15 11:58:07 [TRACE] Graph after step *terraform.ReferenceTransformer:

azurerm_eventhub.test_events_v0 - *terraform.NodeValidatableResourceInstance
2018/08/15 11:58:07 [TRACE] Graph after step *terraform.RootTransformer:

azurerm_eventhub.test_events_v0 - *terraform.NodeValidatableResourceInstance
2018/08/15 11:58:07 [ERROR] root: eval: *terraform.EvalValidateResource, err: Warnings: []. Errors: [invalid value for name (The namespace name can contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.)]
2018/08/15 11:58:07 [ERROR] root: eval: *terraform.EvalSequence, err: Warnings: []. Errors: [invalid value for name (The namespace name can contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.)]
2018/08/15 11:58:07 [TRACE] [walkValidate] Exiting eval tree: azurerm_eventhub.test_events_v0
2018/08/15 11:58:07 [DEBUG] plugin: waiting for all plugin processes to complete...

Error: azurerm_eventhub.test_events_v0: invalid value for name (The namespace name can contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.)


2018-08-15T11:58:07.744+1000 [DEBUG] plugin: plugin process exited: path=/Users/ducas/Code/EventHub/test/.terraform/plugins/darwin_amd64/terraform-provider-azurerm_v1.12.0_x4
```

### Panic Output

N/A

### Expected Behavior

Hub created successfully as per portal and provider version 1.9.0.

### Actual Behavior

Error: module.test_hub_secondary.azurerm_eventhub.test_events_v0: invalid value for name (The namespace name can contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.)

### Steps to Reproduce

1. `terraform init`
2. `terraform validate` => error
3. Change provider to version 1.9.0
4. `terraform init`
5. `terraform validate` => success

### Important Factoids

N/A

### References

N/A

--- a/azurerm/helpers/azure/eventhub.go
+++ b/azurerm/helpers/azure/eventhub.go
@@ -15,13 +15,13 @@ import (
 func ValidateEventHubNamespaceName() schema.SchemaValidateFunc {
 	return validation.StringMatch(
 		regexp.MustCompile("^[a-zA-Z][-a-zA-Z0-9]{4,48}[a-zA-Z0-9]$"),
-		"The namespace name can contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.",
+		"The namespace name can contain only letters, numbers, underscores and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.",
 	)
 }
 
 func ValidateEventHubName() schema.SchemaValidateFunc {
 	return validation.StringMatch(
-		regexp.MustCompile("^[a-zA-Z][-a-zA-Z0-9]{4,48}[a-zA-Z0-9]$"),
+		regexp.MustCompile("^[a-zA-Z][-_a-zA-Z0-9]{4,48}[a-zA-Z0-9]$"),
 		"The namespace name can contain only letters, numbers, and hyphens. The namespace must start with a letter, and it must end with a letter or number and be between 6 and 50 characters long.",
 	)
 }

